In [5]:
import pandas as pd
import pymc as pm
import numpy as np
import arviz as az

file_path = "real_estate_data.csv"
data = pd.read_csv(file_path)

# x1 - suprafata
# x2 - nr camere
# x3 - distanta
# y - pret

x1 = data.iloc[:, 0]
x2 = data.iloc[:, 1]
x3 = data.iloc[:, 2]
y = data.iloc[:, 3]

with pm.Model() as model:
    intercept = pm.Normal("intercept", mu=0, sigma=10)
    beta_x1 = pm.Normal("beta_x1", mu=0, sigma=10)
    beta_x2 = pm.Normal("beta_x2", mu=0, sigma=10)
    beta_x3 = pm.Normal("beta_x3", mu=0, sigma=10)
    sigma = pm.HalfNormal("sigma", sigma=10)

    mu = (intercept +
          beta_x1 * x1 +
          beta_x2 * x2 +
          beta_x3 * x3)

    y_obs = pm.Normal("y_obs", mu=mu, sigma=sigma, observed=y)

    trace = pm.sample(2000, tune=1000, target_accept=0.9)

az.summary(trace, hdi_prob=0.95)


Output()

Output()

mean     sd  hdi_2.5%  hdi_97.5%  mcse_mean  mcse_sd  ess_bulk  \
beta_x1     2.480  0.018     2.445      2.516      0.000    0.000    2128.0   
beta_x2     9.553  0.469     8.647     10.474      0.010    0.007    2340.0   
beta_x3    -3.021  0.196    -3.407     -2.632      0.005    0.003    1868.0   
intercept  33.112  2.533    28.265     37.935      0.064    0.045    1567.0   
sigma       5.540  0.418     4.741      6.384      0.008    0.006    2807.0   

           ess_tail  r_hat  
beta_x1      2503.0    1.0  
beta_x2      2473.0    1.0  
beta_x3      2388.0    1.0  
intercept    2246.0    1.0  
sigma        2200.0    1.0

Din rezultatele obtinute, putem deduce ca cel mai mare impact il are numarul de camere cu beta_2 = 8.647, dupa distanta fata de centru cu beta_3 = -3.407, dupa suprafata cu beta_1 = 2.445.<br>
Putem interpreta ca pentru fiecare unitate cu care crestea valoarea atributului (1 camera in plus, 1 km mai departe de centru sau 1 mp mai mult), valoarea pretului se va modifica cu beta unitati (o crestere in pret de 8647, -3407, respectiv 2445 euro).<br>
<br>
Coeficientul negativ al distantei se explica prin faptul ca un apartament este mai valoros atunci cand distanta fata de centrul orasului este mai mica. Asadar, cu cat distanta este mai mare, cu atat pretul trebuie sa scada.